### PR TITLE
fix:update azurerm version

### PR DESCRIPTION
**Pull request by vjdbj**
## what
* Added `subscription_id` for provider in root module
* Also updated the modules release version in root modules
* Fixed automerge workflow. 

## why
* subscription_id is compulsory to add for terraform planning

### CODE REVIEW

File: README.yaml [hunk 1]
```diff
@@ -19,20 +18,40 @@ badges:
   - name: "Terraform"
     image: "https://img.shields.io/badge/Terraform-v1.1.7-green"
     url: "https://www.terraform.io"
+  - name: "tfsec"
+    image: "https://github.com/clouddrove/terraform-azure-log-analytics/actions/workflows/tfsec.yml/badge.svg"
```
Comment (13archit) [inline README.yaml:22]: log analytics?

File: examples/complete/versions.tf [hunk 2]
```diff
@@ -7,7 +7,11 @@ terraform {
   required_providers {
     azurerm = {
       source  = "hashicorp/azurerm"
-      version = "3.112.0"
+      version = ">=3.112.0"
+    }
+    random = {
+      source  = "hashicorp/random"
+      version = "~> 3.1"
```
Comment (13archit) [inline examples/complete/versions.tf:14]: Can we make it `>=` ?

File: main.tf [hunk 3]
```diff
@@ -102,19 +100,10 @@ resource "azurerm_mysql_flexible_server_configuration" "main" {
...
   resource_group_name = local.resource_group_name
-  server_name         = join("", azurerm_mysql_flexible_server.main.*.name)
+  server_name         = azurerm_mysql_flexible_server.main[0].name
   value               = element(var.values, count.index)
 }
 
-##------------------------------------------------------------------------
-## Manages a Customer Managed Key for a MySQL Server. - Default is "false"
-##------------------------------------------------------------------------
-resource "azurerm_mysql_server_key" "main" {
```
Comment (13archit) [inline main.tf:112]: Why are we removing this, if it is required are we adding any resource/attribute in replace of this?

File: variables.tf [hunk 4]
```diff
@@ -49,11 +50,6 @@ variable "existing_private_dns_zone" {
   default     = false
 }
 
-variable "enable_private_endpoint" {
```
Comment (13archit) [inline variables.tf:52]: Why are we removing this?

File: versions.tf [hunk 5]
```diff
@@ -7,7 +7,11 @@ terraform {
   required_providers {
     azurerm = {
       source  = "hashicorp/azurerm"
-      version = ">=3.0.0"
+      version = ">=3.112.0"
+    }
+    random = {
+      source  = "hashicorp/random"
+      version = "~> 3.1"
```
Comment (13archit) [inline versions.tf:14]: Can we use `>=` here?

Comment (vjdbj) [reply versions.tf:14]: done

File: variables.tf (hunk 4 above)
Comment (vjdbj) [reply variables.tf:52]: It is not being used.

File: main.tf (hunk 3 above)
Comment (vjdbj) [reply main.tf:112]: Added customer_managed_key in azurerm_mysql_flexible_server

File: README.yaml (hunk 1 above)
Comment (vjdbj) [reply README.yaml:22]: corrected

File: examples/complete/versions.tf (hunk 2 above)
Comment (vjdbj) [reply examples/complete/versions.tf:14]: done

File: README.yaml [hunk 6]
```diff
@@ -19,20 +18,40 @@ badges:
...
+    url: "https://github.com/clouddrove/terraform-azure-log-analytics/actions/workflows/tfsec.yml"
```
[the remark this answers is not in the document]
Comment (vjdbj) [reply README.yaml:23]: corrected

File: examples/complete/main.tf [hunk 7]
```diff
@@ -1,14 +1,23 @@
...
+  alias           = "peer"
+  subscription_id = "000000-11111-1223-XXX-XXXXXXXXXXXX"
+}
+
+data "azurerm_client_config" "current_client_config" {}
+
 locals {
-  name        = "app"
-  environment = "test"
+  name        = "lacoster-23"
```
Comment (13archit) [inline examples/complete/main.tf:15]: Please revert it to `app` and `test`

File: examples/complete/main.tf [hunk 8]
```diff
@@ -60,12 +69,56 @@ module "subnet" {
...
+module "vault" {
+  source  = "clouddrove/key-vault/azure"
+  version = "1.2.0"
+
+  providers = {
+    azurerm.dns_sub  = azurerm.peer,
+    azurerm.main_sub = azurerm
+  }
+
+  name                        = "oliveware-23"
```
Comment (13archit) [inline examples/complete/main.tf:84]: Please use local here in name and environment.

File: examples/complete/main.tf (hunk 7 above)
Comment (vjdbj) [reply examples/complete/main.tf:15]: done

File: examples/complete/main.tf (hunk 8 above)
Comment (vjdbj) [reply examples/complete/main.tf:84]: done

File: examples/mysql-flexible-server-replication/main.tf [hunk 9]
```diff
@@ -61,29 +61,29 @@ module "subnet" {
...
   resource_group_name = data.azurerm_resource_group.main.name
 }
 
-##----------------------------------------------------------------------------- 
+##-----------------------------------------------------------------------------
 ## Flexible Mysql server module call.
 ##-----------------------------------------------------------------------------
 module "flexible-mysql" {
   depends_on                     = [module.resource_group, module.vnet, data.azurerm_resource_group.main]
-  source                         = "clouddrove/flexible-mysql/azure"
```
Comment (d4kverma) [inline examples/mysql-flexible-server-replication/main.tf:86]: add source here

File: examples/mysql-flexible-server-replication/main.tf [hunk 10]
```diff
@@ -92,7 +92,6 @@ module "flexible-mysql" {
   virtual_network_id             = module.vnet.vnet_id[0]
   delegated_subnet_id            = module.subnet.default_subnet_id[0]
   mysql_version                  = "8.0.21"
-  mysql_server_name              = "testmysqlserver"
```
Comment (d4kverma) [inline examples/mysql-flexible-server-replication/main.tf:95]: any specific reason to remove it

File: examples/mysql-flexible-server-replication/main.tf (hunk 9 above)
Comment (vjdbj) [reply examples/mysql-flexible-server-replication/main.tf:86]: Screenshot from 2024-11-05 16-06-14

File: examples/mysql-flexible-server-replication/main.tf (hunk 10 above)
Comment (vjdbj) [reply examples/mysql-flexible-server-replication/main.tf:95]: tflint gave error that this variable is not being use other examples. 
that's why I removed this variable and we can give the name of it from labels.

**Pull request merged**